liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/BasicSupport.java

```diff
@@ -198,17 +189,40 @@ protected void init() throws MojoExecutionException, MojoFailureException {
                 installType = InstallType.FROM_FILE;
                 installDirectory = checkServerHome(assemblyArchive);
                 log.info(MessageFormat.format(messages.getString("info.variable.set"), "installDirectory", installDirectory));
-            } else if (assemblyArtifact != null) {
+            } else if(install != null) {
+                installType = InstallType.FROM_ARCHIVE;
+                installDirectory = new File(assemblyInstallDirectory, "wlp");
+                log.info(MessageFormat.format(messages.getString("info.variable.set"), "installDirectory", installDirectory));
+            }
+            else { // default to install from runtime artifact
+                assemblyArtifact.setType("zip");
+                if(assemblyArtifact.getGroupId() == null) {
+                    log.debug("Defaulting runtimeArtifact group id to 'io.openliberty'");
+                    assemblyArtifact.setGroupId("io.openliberty");
+                }
+                if(assemblyArtifact.getArtifactId() == null) {
```

Comment:
What if the groupId was already set but not the artifactId, or vice versa? We could end up with a bad combination here. Seems like we should set both or neither in that case, and log a message.

Reply:
We could end up with a bad combination, but if we document what the defaults are then someone will know if they need to set both a group id and artifact id. I think having a default here is better.